src/hierarchy_builder.rs

```diff
@@ -0,0 +1,206 @@
+extern crate geo;
+
+use zone::{Zone, ZoneIndex};
+use gst::rtree::{RTree, Rect};
+use ordered_float::OrderedFloat;
+use geo::Bbox;
+use geo::boundingbox::BoundingBox;
+
+pub struct ZonesTree {
+    pub tree: RTree<ZoneIndex>,
+}
+
+impl ZonesTree {
+    pub fn new() -> ZonesTree {
+        ZonesTree { tree: RTree::new() }
+    }
+
+    pub fn insert_zone(&mut self, z: &Zone) {
+        match &z.boundary {
```

Comment:
really minor, but this can be written:
```
if let Some(ref b) = z.boundary {
...
}
```

And I think it's better to remove the unwrap (at least an `expect` if we're sure never to encounter the error, else maybe just log it and do nothing ?)